### ISSUE TITLE
Crash when starting the VPN

### ISSUE THREAD

**Issue by pacien** [pacien/tincapp#100]
Issue: the application crashes when starting the VPN, right after selecting a network or when trying to resume operations after the device has been restarted.

__Please comment with more details if you experience this issue!__

---

Automatic crash report from Android Vitals received on 2020-09-18.

* Affected app versions:
  * v0.31
* Affected platforms:
  * Android 10 (Samsung)

Stack trace:
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4552)
  at android.app.ActivityThread.access$2100 (ActivityThread.java:274)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2134)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8154)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
Caused by: java.io.IOException: 
  at android.net.LocalSocketImpl.bindLocal (LocalSocketImpl.java)
  at android.net.LocalSocketImpl.bind (LocalSocketImpl.java:312)
  at android.net.LocalServerSocket.<init> (LocalServerSocket.java:49)
  at org.pacien.tincapp.service.TincVpnService.startVpn (TincVpnService.java:148)
  at org.pacien.tincapp.service.TincVpnService.onStartCommand (TincVpnService.java:66)
  at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4534)
  at android.app.ActivityThread.access$2100 (ActivityThread.java:274)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2134)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8154)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
```

**Comment by fatih2424** [comment]
> Issue: the application crashes when starting the VPN, right after selecting a network or when trying to resume operations after the device has been restarted.
> 
> __Please comment with more details if you experience this issue!__
> 
> ---
> 
> Automatic crash report from Android Vitals received on 2020-09-18.
> 
> * Affected app versions:
>   * v0.31
> * Affected platforms:
>   * Android 10 (Samsung)
> 
> Stack trace:
> ```
> java.lang.RuntimeException: 
>   at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4552)
>   at android.app.ActivityThread.access$2100 (ActivityThread.java:274)
>   at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2134)
>   at android.os.Handler.dispatchMessage (Handler.java:107)
>   at android.os.Looper.loop (Looper.java:237)
>   at android.app.ActivityThread.main (ActivityThread.java:8154)
>   at java.lang.reflect.Method.invoke (Method.java)
>   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
>   at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
> Caused by: java.io.IOException: 
>   at android.net.LocalSocketImpl.bindLocal (LocalSocketImpl.java)
>   at android.net.LocalSocketImpl.bind (LocalSocketImpl.java:312)
>   at android.net.LocalServerSocket.<init> (LocalServerSocket.java:49)
>   at org.pacien.tincapp.service.TincVpnService.startVpn (TincVpnService.java:148)
>   at org.pacien.tincapp.service.TincVpnService.onStartCommand (TincVpnService.java:66)
>   at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4534)
>   at android.app.ActivityThread.access$2100 (ActivityThread.java:274)
>   at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2134)
>   at android.os.Handler.dispatchMessage (Handler.java:107)
>   at android.os.Looper.loop (Looper.java:237)
>   at android.app.ActivityThread.main (ActivityThread.java:8154)
>   at java.lang.reflect.Method.invoke (Method.java)
>   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
>   at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
> ```

**Comment by fatih2424** [comment]
33cad47025041ca053a69ed74a4d0e0446af1550

**Comment by pacien** [comment]
This doesn't show anymore in the Google Play Console.

I've seen a similar stacktrace when double-taping on a network too rapidly too.
This should be fixed in 202e741ef92d102646dd40b9a1bdae97bc095318.
